x-pack/test/security_solution_api_integration/test_suites/detections_response/utils/rules/prebuilt_rules/delete_all_prebuilt_rule_assets.ts

```diff
@@ -4,20 +4,26 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-
+import type { ToolingLog } from '@kbn/tooling-log';
 import type { Client } from '@elastic/elasticsearch';
 import { SECURITY_SOLUTION_SAVED_OBJECT_INDEX } from '@kbn/core-saved-objects-server';
+import { retryIfConflicts } from '../../retry_if_conflicts';
 
 /**
  * Remove all prebuilt rule assets from the security solution savedObjects index
  * @param es The ElasticSearch handle
  */
-export const deleteAllPrebuiltRuleAssets = async (es: Client): Promise<void> => {
-  await es.deleteByQuery({
-    index: SECURITY_SOLUTION_SAVED_OBJECT_INDEX,
-    q: 'type:security-rule',
-    wait_for_completion: true,
-    refresh: true,
-    body: {},
+export const deleteAllPrebuiltRuleAssets = async (
+  es: Client,
+  logger: ToolingLog
+): Promise<void> => {
+  await retryIfConflicts(logger, 'deleteAllPrebuiltRuleAssets', async () => {
```

Comment:
Nit: `'deleteAllPrebuiltRuleAssets'` could probably be replaced by `deleteAllPrebuiltRuleAssets.name` which would be refactoring-safe.